Tasks/PublishTestResults/publishtestresults.ts

```diff
@@ -30,10 +30,11 @@ else {
   var matchingTestResultsFiles = [testResultsFiles];
 }
 
-if(!matchingTestResultsFiles) {
+if(!matchingTestResultsFiles || matchingTestResultsFiles.length == 0) {
   tl.warning('No test result files matching ' + testResultsFiles + ' were found.');  
   tl.exit(0);
 }
-
-var tp = new tl.TestPublisher(testRunner);
-tp.publish(matchingTestResultsFiles, mergeResults, platform, config, testRunTitle, publishRunAttachments);
\ No newline at end of file
+else{
```

Comment:
Note that tl.exit is deprecated but supported for back compat.  Abrubtly exiting the script (process.exit) can cause some output to be lost.  I will remove in next majoor update of task lib.  Just let the task exit and use tl.setresult to success or failure.